app/icons/index.tsx

```diff
@@ -44,3 +53,23 @@ export const getChartIcon = (chartType: ChartType): IconName => {
       return "table";
   }
 };
+
+export const getDimensionIconName = (
+  dimensionType: ComponentType
+): IconName => {
+  switch (dimensionType) {
+    case "GeoCoordinatesDimension":
+    case "GeoShapesDimension":
+      return "geographical";
+    case "NominalDimension":
+      return "chartPie";
+    case "NumericalMeasure":
+      return "numerical";
+    case "OrdinalDimension":
+      return "sort";
+    case "OrdinalMeasure":
+      return "sort";
+    case "TemporalDimension":
+      return "pointInTime";
+  }
```

Comment:
could we have a exhaustive check here ?

I think `return dimensionType as never` should work.